src/System.Net.WebClient/tests/WebClientTest.cs

```diff
@@ -54,23 +53,29 @@ public static void DownloadData_InvalidArguments_ThrowExceptions()
             AssertExtensions.Throws<ArgumentNullException>("address", () => { wc.DownloadDataAsync((Uri)null); });
             AssertExtensions.Throws<ArgumentNullException>("address", () => { wc.DownloadDataAsync((Uri)null, null); });
 
-            AssertExtensions.Throws<ArgumentNullException>("address", () => { wc.DownloadDataTaskAsync((string)null); });
+            if (!PlatformDetection.IsFullFramework) // netfx has bug and returns wrong argument name in exception.
+            {
+                AssertExtensions.Throws<ArgumentNullException>("address", () => { wc.DownloadDataTaskAsync((string)null); });
```

Comment:
AssertExtensions have a method that receives both the expected param name for NETFx and Net Core and it does the runtime check and compares the param name to the corresponding expected name, so instead of doing the check here should we call it with NETFx and Net Core expected param names?